propka/output.py

```diff
@@ -1,397 +1,491 @@
-    """
-    writing a scwrl sequence file for, e.g.,  generating a mutated protein
+    TODO - figure out what this is
     """
-    file = open(filename, 'w')
-
-    start = 0
-    while len(sequence[start:]) > 60:
-      file.write( "%s\n" % (sequence[start:start+60]) )
-      start += 60
-    file.write( "%s\n" % (sequence[start:]) )
+    with open(filename, 'w') as file_:
+        start = 0
+        while len(sequence[start:]) > 60:
+            file_.write("{0:s}s\n".format(sequence[start:start+60]))
+            start += 60
+        file_.write("{0:s}\n".format(sequence[start:]))
 
-    file.close()
 
+def get_propka_header():
+    """Create the header.
 
-
-# --- various header text --- #
-
-
-def getPropkaHeader():
+    Returns:
+        string
     """
-    Creates the header
-    """
-    from datetime import date
     today = date.today()
-
-
-
-    str  = "propka3.1 %93s\n" % (today)
-    str += "-------------------------------------------------------------------------------------------------------\n"
-    str += "--                                                                                                   --\n"
-    str += "--                                   PROPKA: A PROTEIN PKA PREDICTOR                                 --\n"
-    str += "--                                                                                                   --\n"
-    str += "--                                 VERSION 1.0,  04/25/2004, IOWA CITY                               --\n"
-    str += "--                                             BY HUI LI                                             --\n"
-    str += "--                                                                                                   --\n"
-    str += "--                            VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                         --\n"
-    str += "--                                BY DELPHINE C. BAS AND DAVID M. ROGERS                             --\n"
-    str += "--                                                                                                   --\n"
-    str += "--                                VERSION 3.0,  01/06/2011, COPENHAGEN                               --\n"
-    str += "--                            BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                         --\n"
-    str += "--                                                                                                   --\n"
-    str += "--                                VERSION 3.1,  07/01/2011, COPENHAGEN                               --\n"
-    str += "--                            BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                         --\n"
-    str += "-------------------------------------------------------------------------------------------------------\n"
-    str += "\n"
-
-
-
-    return str
-
-
-def getReferencesHeader():
+    str_ = "propka3.1 {0:>93s}\n".format(today)
```

Comment:
This needs `!s` string conversion: `"propka3.1 {0!s:>93}".format(today)`

Reply:
Addressed in https://github.com/Electrostatics/propka-3.1/commit/b54190ef279b9acd98f6b52cfc861be14fd94e84